src/main/java/com/eleks/auctioneer/service/UserService.java

```diff
@@ -0,0 +1,34 @@
+package com.eleks.auctioneer.service;
+
+import com.eleks.auctioneer.DTO.AppUserDTO;
+import com.eleks.auctioneer.entity.AppUser;
+import com.eleks.auctioneer.repository.UserInfoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserService {
+    @Autowired
+    private UserInfoRepository userInfoRepository;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    public String addUser(AppUserDTO userInfoDTO) {
+        if(userInfoRepository.findByEmail(userInfoDTO.getEmail()).isEmpty())
+        {
+            userInfoDTO.setPassword(passwordEncoder.encode(userInfoDTO.getPassword()));
```

Comment:
It's better to do the encoding inside the "mapToUser" method. Just pass PasswordEncoder as a second parameter.